icicle/appUtils/msm/msm.cu

```diff
@@ -427,6 +427,7 @@ void bucket_method_msm(unsigned bitsize, unsigned c, S *scalars, A *points, unsi
   unsigned bucket_th = large_bucket_factor*avarage_size;
   unsigned *nof_large_buckets;
   cudaMallocAsync(&nof_large_buckets, sizeof(unsigned), stream);
+  cudaMemset(nof_large_buckets, 0, sizeof(unsigned));
```

Comment:
why is this needed?

Reply:
I don't really see how the case of zero large buckets is handled. If there are no large buckets, `find_cutoff_kernel` just doesn't modify the `nof_large_buckets` variable, so it stays whatever it was when allocated on line 429. Because `cudaMallocAsync` [doesn't clear](https://stackoverflow.com/a/62055960) the memory, we potentially  might get gibberish instead of zero in `nof_large_buckets`. Admittedly, I never actually got gibberish but I think it's better to do it the safe way